packages/back-end/src/controllers/vercel.ts

```diff
@@ -0,0 +1,153 @@
+import fetch from "node-fetch";
+import { Response } from "express";
+import { AuthRequest } from "../types/AuthRequest";
+import { updateOrganization } from "../models/OrganizationModel";
+import { getOrgFromReq } from "../services/organizations";
+import { getAllApiKeysByOrganization } from "../services/apiKey";
+import {
+  GbVercelEnvMap,
+  ApiKeyRow,
+  VercelEnvVar,
+  CreateEnvParams,
+} from "../../types/vercel";
+import {
+  createOrgGbKeys,
+  getEnvVars,
+  getGbRelatedVercelProjects,
+  postEnvVar,
+  reduceGbVercelEnvMap,
+} from "../services/vercel";
+import { VERCEL_CLIENT_ID, VERCEL_CLIENT_SECRET } from "../util/secrets";
+import { auditDetailsUpdate } from "../services/audit";
+
+export async function getHasToken(req: AuthRequest, res: Response) {
+  const { org } = await getOrgFromReq(req);
+  res.status(200).json({ hasToken: !!org.connections?.vercel?.token });
+}
+
+export async function postToken(
+  req: AuthRequest<{ code: string; configurationId: string; teamId: string }>,
+  res: Response
+) {
+  req.checkPermissions("organizationSettings");
+  const { code, configurationId, teamId } = req.body;
+  const { org } = getOrgFromReq(req);
+
+  const url = "https://api.vercel.com/v2/oauth/access_token";
+  const options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded",
+    },
+    body: new URLSearchParams({
+      code: code,
+      client_id: VERCEL_CLIENT_ID,
+      client_secret: VERCEL_CLIENT_SECRET,
+      redirect_uri: "https://app.growthbook.io/integrations/vercel",
+    }),
+  };
+
+  const tokenRes = await fetch(url, options);
+  const json = await tokenRes.json();
+  if (json.error) throw new Error(json.error);
+
+  const orig = org.connections?.vercel;
+  const updatePayload = { token: json.access_token, configurationId, teamId };
+
+  await updateOrganization(org.id, { connections: { vercel: updatePayload } });
+
+  await req.audit({
+    event: "organization.update",
+    entity: {
+      object: "organization",
+      id: org.id,
+    },
+    details: auditDetailsUpdate(
+      { connections: { vercel: orig } },
+      { connections: { vercel: updatePayload } }
+    ),
```

Comment:
I don't think we want to store the Vercel token in the audit log in plain text.  That's likely to get leaked to the front-end at some point.  The important thing to have in the audit log is the teamId and configurationId.  Maybe we can just replace the token with `*********` for the audit details.